graveler/staging_test.go

```diff
@@ -118,6 +116,158 @@ func TestDrop(t *testing.T) {
 	}
 }
 
+func TestDropByPrefix(t *testing.T) {
+	s := newTestStagingManager(t)
+	numOfValues := 2400
+	for i := 0; i < numOfValues; i++ {
+		err := s.Set(context.Background(), "t1", []byte(fmt.Sprintf("key%04d", i)), newTestValue(fmt.Sprintf("identity%d", i), fmt.Sprintf("value%d", i)))
+		if err != nil {
+			t.Fatalf("got unexpected error: %v", err)
+		}
+		err = s.Set(context.Background(), "t2", []byte(fmt.Sprintf("key%04d", i)), newTestValue(fmt.Sprintf("identity%d", i), fmt.Sprintf("value%d", i)))
+		if err != nil {
+			t.Fatalf("got unexpected error: %v", err)
+		}
+	}
+	err := s.DropByPrefix(context.Background(), "t1", []byte("key1"))
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	v, err := s.Get(context.Background(), "t1", []byte("key1000"))
+	if !errors.Is(err, db.ErrNotFound) {
+		// key1000 starts with the deleted prefix - should have been deleted
+		t.Fatalf("after dropping staging area, expected ErrNotFound in Get. got err=%v, got value=%v", err, v)
+	}
+	v, err = s.Get(context.Background(), "t1", []byte("key0000"))
+	if err != nil {
+		// key0000 does not start with the deleted prefix - should be returned
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	it, _ := s.List(context.Background(), "t1")
+	count := 0
+	for it.Next() {
+		count++
+	}
+	it.Close()
```

Comment:
extract method? also can check the error or extract smaller scope and have count by iter that will use anything with Next to return length